### PR TITLE
rename various things to be consistent with net-* components

### PR DESCRIPTION
**Pull request by dprotaso**
Part of https://github.com/knative/networking/issues/448

### CODE REVIEW

File: config/200-serviceaccount.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: 3scale-kourier
+  name: net-kourier
```
Comment (dprotaso) [inline config/200-serviceaccount.yaml:18]: Curious any reason why you don't opt into the aggregated cluster role?

File: config/300-controller.yaml [hunk 2]
```diff
@@ -24,15 +24,15 @@ spec:
...
   template:
     metadata:
       labels:
-        app: 3scale-kourier-control
+        app: net-kourier-controller
     spec:
       containers:
         - image: ko://knative.dev/net-kourier/cmd/kourier
-          name: kourier-control
+          name: controller
```
Comment (nak3) [inline config/300-controller.yaml:35]: I'm fine for this change, but users have to change `spec.registry.override` if they configured :sweat_drops: . In addition, they need to change it `deployment/container` style like  https://github.com/knative/docs/pull/3821/files

File: config/200-serviceaccount.yaml (hunk 1 above)
Comment (nak3) [reply config/200-serviceaccount.yaml:18]: I think no reason. We can use aggregated cluster role.

**Comment by nak3** [comment]
LGTM

**Comment by dprotaso** [comment]
So the gateway deployment is renamed but the labels will remain the same

**Comment by dprotaso** [comment]
@nak3 @markusthoemmes this is ready now

File: pkg/reconciler/ingress/controller.go [hunk 3]
```diff
@@ -51,9 +51,9 @@ import (
 
 const (
 	gatewayLabelKey   = "app"
-	gatewayLabelValue = "3scale-kourier-gateway"
+	gatewayLabelValue = "net-kourier-gateway"
```
Comment (markusthoemmes) [inline pkg/reconciler/ingress/controller.go:54]: This is wrong I believe, as you haven't changed the label for the gateways.

**Review by markusthoemmes — COMMENTED** [review]
Verified this works as intended locally. Verified that the config change does not impact the running cluster.

**Comment by markusthoemmes** [comment]
Gonna hold this for this release. Manual application has been verified but I'd really like the operator to behave nicely with the changes as well. Currently, that's not been verified and I'm quite sure it breaks in that traffic will be dropped. I'll attend to this PR more closely to target the next release for landing. Sorry for not getting here earlier.

/hold

**Review by markusthoemmes — APPROVED** [review]
/lgtm
/approve
/unhold

**Pull request merged**